### ISSUE TITLE
[CT-2033] Move all tests into `tests/`, and ensure pre-commit checks are turned on for tests

### ISSUE THREAD

**Issue by dbeatty10** [dbt-labs/dbt-core#6862]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Caveat: I did a _quick_ search of existing issues, so it is possible there is a pre-existing issue that wasn't able to catch with my search terms.

There is a comment within `.pre-commit-config.yaml` that exists across repos, and it most commonly looks like:
```python
# TODO: remove global exclusion of tests when testing overhaul is complete
exclude: '^tests/.*'
```

So once the testing overhaul is complete, it seems like we should... remove these lines as mentioned in the TODO.

### Acceptance criteria

The following acceptance criteria applies to dbt-core + dbt-* adapters:
- [x] Remove `exclude: '^test/.*'` exclusion (and its `# TODO` comment)
- [ ] Remove the [`interop` directory](https://github.com/dbt-labs/dbt-core/tree/main/test/interop)
  - [ ] Afterwards, confirm that we're still running the structured logging test somehow
  - [ ] Remove customizations within the `Makefile` related to the `interop` directory code (e.g., https://github.com/dbt-labs/dbt-core/pull/5763)
- [ ] Move all remaining tests into the `tests/` directory
- [ ] Delete the (now empty) `test` directory

### Instances
- https://github.com/dbt-labs/dbt-core/blob/b2ea2b8b256e5db1da0b712dfedd7973e1e50a37/.pre-commit-config.yaml#L4-L5
- https://github.com/dbt-labs/dbt-bigquery/blob/4fc8efd88942358b618ceb5c87416b19a29f4ed4/.pre-commit-config.yaml#L3-L4
- https://github.com/dbt-labs/dbt-redshift/blob/28d22bd8abf53882c7f4ddde6d4b611a5c924a9d/.pre-commit-config.yaml#L3-L4
- https://github.com/dbt-labs/dbt-snowflake/blob/5c8410b22592a0a248d44629580c5de331196d4f/.pre-commit-config.yaml#L3-L4
- https://github.com/dbt-labs/dbt-spark/blob/55d86d09b8630b2e0a2107c696d632721fd04bd4/.pre-commit-config.yaml#L3-L4

**Comment by gshank** [comment]
There should never have been an exclusion of tests/ in dbt-core. The intention was that "test/" was excluded. There was, for a while, a typo introduced in dbt-core that excluded 'test' (without the following slash) which excluded tests accidentally. The comment probably dated from initial setup, when we wanted to not reformat everything in test/, but intentionally include everything in tests/.  The adapter repos had "tests" directories instead, so it wasn't as easy to just exclude the non-converted tests.  At this point, we could probably just remove the exclusion and reformat everything. (Though it would also be nice to move test/unit to tests/unit in dbt-core.) I think when we originally set the exclusion up, the fear was that with all of the test directories reformatted at that point it would make it difficult to backport fixes. Right now is probably as good as it gets for avoiding backporting issues.

**Comment by jtcohen6** [comment]
Agreed, we should do this now!

Any opposition to moving all remaining tests into the `tests/` directory? The remaining tests in there are:
- unit tests (not yet converted, but not a priority right now)
- structured logging schema check (should we be doing this by other means? https://github.com/dbt-labs/dbt-core/issues/6852)

**Comment by gshank** [comment]
I think it would be fine to move them all into tests. Would be nice to get rid of the "test" directory!  I don't believe that we're currently using the "interop" directory code, so we can probably remove it. We're still running the structured logging test, but it's not using that code. If we wanted to bring it back in the future, git history is forever :)

**Comment by emmyoop** [comment]
This is a relatively low lift now since the tests in `test/` are being checked for formatting now.  It's should just be a matter of moving them without the overhead of fixing them.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues this week — I think this one has shipped and can be closed.

**Evidence (verified against current `main`):**

- [`tests/`](https://github.com/dbt-labs/dbt-core/blob/main/tests/) — All tests live under tests/ now (conftest.py, data/, fixtures/, functional/, unit/) — no top-level `test/` directory exists
- `.pre-commit-config.yaml` — global exclude is `^(core/dbt/docs/build/|core/dbt/common/events/types_pb2.py|core/dbt/adapters/events/adapter_types_pb2.py)` — no longer excludes `^tests/.*`

The acceptance criteria from this issue have been completed: tests directory unified, pre-commit no longer skips tests, old `test/` directory removed.

Would you mind closing this out? If something specific is still missing — a sub-field, a related behavior — point me at it and I'll dig further.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually against the current `main` branch.)
